Note that _building_ and _testing_ are two different things; the _build_ succeeds, but one of the _tests_ fails. 

When you invoke `utils/build-script` _without_ the `--test` flag, it will only build.
When you _do_ pass the `--test` flag, it will build and then also run the tests for you. 

You can also run the tests without first building everything; see https://github.com/apple/swift/blob/main/docs/HowToGuides/GettingStarted.md#running-tests.
Note that this way lets you use `--filter=racy_async_let_fibonacci` to run this particular test in isolation.

I did that just now, quite a number of times, but now it fails to fail (it passes)...

The only somewhat related log output is the following. Seems not to contain anything valuable. 
<details><summary>Show</summary>

```
+ /opt/homebrew/opt/python@3.9/bin/python3.9 ../llvm-project/llvm/utils/lit/lit.py -sv --param swift_test_mode=optimize_none --param swift_test_subset=primary /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/test-macosx-arm64
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:216: note: Using cmake: /opt/homebrew/Cellar/cmake/3.22.2/bin/cmake
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-frontend: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-frontend
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swiftc: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swiftc
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using sil-opt: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/sil-opt
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using sil-func-extractor: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/sil-func-extractor
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using sil-llvm-gen: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/sil-llvm-gen
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using sil-nm: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/sil-nm
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using sil-passpipeline-dumper: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/sil-passpipeline-dumper
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using lldb-moduleimport-test: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/lldb-moduleimport-test
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-ide-test: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-ide-test
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-dependency-tool: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-dependency-tool
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-syntax-test: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-syntax-test
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-syntax-parser-test: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-syntax-parser-test
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-reflection-dump: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-reflection-dump
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-remoteast-test: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-remoteast-test
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-indent: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-indent
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-symbolgraph-extract: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-symbolgraph-extract
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-api-extract: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-api-extract
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using clang: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/clang
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using clang++: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/clang++
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using llvm-link: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/llvm-link
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-llvm-opt: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-llvm-opt
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using llvm-profdata: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/llvm-profdata
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using llvm-cov: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/llvm-cov
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using llvm-strings: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/llvm-strings
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using FileCheck: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/FileCheck
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using llvm-dwarfdump: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/llvm-dwarfdump
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using llvm-readelf: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/llvm-readelf
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using llvm-dis: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/llvm-dis
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using llvm-nm: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/llvm-nm
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using llvm-size: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/llvm-size
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using sourcekitd-test: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/sourcekitd-test
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using complete-test: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/complete-test
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-api-digester: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-api-digester
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-refactor: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-refactor
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-demangle-yamldump: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-demangle-yamldump
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-demangle: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-demangle
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:303: warning: couldn't find 'Benchmark_O' program, try setting BENCHMARK_O in your environment
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:303: warning: couldn't find 'Benchmark_Driver' program, try setting BENCHMARK_DRIVER in your environment
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:303: warning: couldn't find 'wasmer' program, try setting WASMER in your environment
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:303: warning: couldn't find 'wasm-ld' program, try setting WASM_LD in your environment
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:382: note: Using resource dir: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/lib/swift
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:429: note: Compiling with -swift-version 4
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:465: note: Using Clang module cache: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/swift-test-results/arm64-apple-macosx10.9/clang-module-cache
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:466: note: Using test results dir: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/./swift-test-results/arm64-apple-macosx10.9
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:471: note: Using code completion cache: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/swift-test-results/arm64-apple-macosx10.9/completion-cache
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:1154: note: Testing OS X arm64-apple-macosx10.9
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:1205: note: Running tests on macOS version 12.2 (21D49)
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:1998: note: Using platform module dir: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/lib/swift/macosx
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:2055: note: Testing with the just-built libraries at /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/lib/swift/macosx
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:300: note: using swift-reflection-test-macosx-arm64: /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-reflection-test-macosx-arm64
lit.py: /Users/casper/Desktop/Swift/swift/test/lit.cfg:2524: note: Available features: CMAKE_GENERATOR=Ninja, CODEGENERATOR=AArch64, CODEGENERATOR=ARM, CODEGENERATOR=Mips, CODEGENERATOR=PowerPC, CODEGENERATOR=SystemZ, CODEGENERATOR=X86, CPU=arm64, OS=macosx, PTRSIZE=64, SWIFT_VERSION=4, VENDOR=apple, asan_runtime, asserts, c_runtime, concurrency, concurrency_runtime, cplusplus_driver, crash-recovery, differentiable_programming, distributed, executable_test, foundation, fuzzer_runtime, libdispatch, libxml2, lld_lto, no_asan, no_lto, nonexecutable_test, objc_interop, optimized_stdlib, optimized_stdlib_arm64, profile_runtime, reflection, reflection_test_support, runtime_function_counters, sftp_server, shell, sourcekit, string_processing, swift-remoteast-test, swift_interpreter, swift_only_stable_abi, swift_repl, swift_stable_abi, swift_stdlib_asserts, swift_test_mode_optimize_none, swift_test_mode_optimize_none_arm64, swift_tools_extra, syntax_parser_lib, tsan_runtime, ubsan_runtime
objc[86267]: Class __SwiftNSErrorLayoutStandin is implemented in both /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/lib/swift/macosx/libswiftCore.dylib (0x103fd6800) and /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/unittests/runtime/SwiftRuntimeTests (0x10274ca18). One of the two will be used. Which one is undefined.
objc[86267]: Class __SwiftNativeNSError is implemented in both /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/lib/swift/macosx/libswiftCore.dylib (0x103fd6828) and /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/unittests/runtime/SwiftRuntimeTests (0x10274ca40). One of the two will be used. Which one is undefined.
objc[86267]: Class _TtCs12_SwiftObject is implemented in both /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/lib/swift/macosx/libswiftCore.dylib (0x103fd6878) and /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/unittests/runtime/SwiftRuntimeTests (0x10274ca90). One of the two will be used. Which one is undefined.
objc[86267]: Class __SwiftValue is implemented in both /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/lib/swift/macosx/libswiftCore.dylib (0x103fd68f0) and /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/unittests/runtime/SwiftRuntimeTests (0x10274cb08). One of the two will be used. Which one is undefined.
FAIL: Swift(macosx-arm64) :: Sanitizers/tsan/racy_async_let_fibonacci.swift (57 of 8259)
******************** TEST 'Swift(macosx-arm64) :: Sanitizers/tsan/racy_async_let_fibonacci.swift' FAILED ********************
Script:
--
: 'RUN: at line 1';   xcrun --toolchain default --sdk '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk' /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swiftc -toolchain-stdlib-rpath -target arm64-apple-macosx10.9  -module-cache-path /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/swift-test-results/arm64-apple-macosx10.9/clang-module-cache -Xlinker -headerpad_max_install_names -Xlinker -rpath -Xlinker /usr/lib/swift  -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 9999:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.0:macOS 10.14.4, iOS 12.2, watchOS 5.2, tvOS 12.2' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.1:macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.2:macOS 10.15.4, iOS 13.4, watchOS 6.2, tvOS 13.4' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.3:macOS 11.0, iOS 14.0, watchOS 7.0, tvOS 14.0' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.4:macOS 11.3, iOS 14.5, watchOS 7.4, tvOS 14.5' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.5:macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.6:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999'  /Users/casper/Desktop/Swift/swift/test/Sanitizers/tsan/racy_async_let_fibonacci.swift -Xfrontend -disable-availability-checking -parse-as-library  -target arm64-apple-macosx10.9 -g -sanitize=thread -o /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/test-macosx-arm64/Sanitizers/tsan/Output/racy_async_let_fibonacci.swift.tmp
: 'RUN: at line 2';   /Users/casper/Desktop/Swift/swift/utils/swift-darwin-postprocess.py /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/test-macosx-arm64/Sanitizers/tsan/Output/racy_async_let_fibonacci.swift.tmp
: 'RUN: at line 3';   env TSAN_OPTIONS="abort_on_error=0" not /usr/bin/env DYLD_LIBRARY_PATH='/Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/lib/swift/macosx' LD_LIBRARY_PATH='/Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/lib/swift/macosx:' SIMCTL_CHILD_DYLD_LIBRARY_PATH='/Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/lib/swift/macosx'  /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/test-macosx-arm64/Sanitizers/tsan/Output/racy_async_let_fibonacci.swift.tmp 2>&1 | /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-demangle --simplified | /opt/homebrew/opt/python@3.9/bin/python3.9 /Users/casper/Desktop/Swift/swift/utils/PathSanitizingFileCheck --allow-unused-prefixes --sanitize BUILD_DIR=/Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64 --sanitize SOURCE_DIR=/Users/casper/Desktop/Swift/swift --use-filecheck /Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/FileCheck  /Users/casper/Desktop/Swift/swift/test/Sanitizers/tsan/racy_async_let_fibonacci.swift
--
Exit Code: 1

Command Output (stderr):
--
/Users/casper/Desktop/Swift/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/test-macosx-arm64/Sanitizers/tsan/Output/racy_async_let_fibonacci.swift.tmp: replacing existing signature
/Users/casper/Desktop/Swift/swift/test/Sanitizers/tsan/racy_async_let_fibonacci.swift:70:11: error: CHECK: expected string not found in input
// CHECK: ThreadSanitizer: Swift access race
          ^
<stdin>:1:1: note: scanning from here

^
<stdin>:2:15: note: possible intended match here
Async fib = 55, sequential fib = 55
              ^

Input file: <stdin>
Check file: /Users/casper/Desktop/Swift/swift/test/Sanitizers/tsan/racy_async_let_fibonacci.swift

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1:  
check:70'0     X error: no match found
            2: Async fib = 55, sequential fib = 55 
check:70'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:70'1                   ?                      possible intended match
>>>>>>

--

```

</details>

--- a/test/Sanitizers/tsan/racy_async_let_fibonacci.swift
+++ b/test/Sanitizers/tsan/racy_async_let_fibonacci.swift
@@ -15,6 +15,7 @@
 // UNSUPPORTED: OS=watchos
 
 // rdar://86825277
+// SR-15805
 // UNSUPPORTED: CPU=arm64 || CPU=arm64e
 
 func fib(_ n: Int) -> Int {

